core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -257,15 +254,21 @@ class KafkaServer(val config: KafkaConfig, time: Time = SystemTime, threadNamePr
 
     if (chroot.length > 1) {
       val zkConnForChrootCreation = config.zkConnect.substring(0, config.zkConnect.indexOf("/"))
-      val zkClientForChrootCreation = ZkUtils.createZkClient(zkConnForChrootCreation, config.zkSessionTimeoutMs, config.zkConnectionTimeoutMs)
-      ZkUtils.makeSurePersistentPathExists(zkClientForChrootCreation, chroot)
+      val zkClientForChrootCreation = ZkUtils.create(zkConnForChrootCreation, 
+                                                     config.zkSessionTimeoutMs,
+                                                     config.zkConnectionTimeoutMs,
+                                                     ToolsUtils.isSecure(System.getProperty("java.security.auth.login.config")))
```

Comment:
In the SASL PR, there is a `org.apache.kafka.common.security.JaasUtils` in the clients jar. Maybe we could use the same file in this PR? That will create a merge conflict, but it will be trivial to resolve and in a single file.